gammapy/cube/make.py

```diff
@@ -134,15 +151,42 @@ def __init__(self, obs, geom, fov_mask=None, exclusion_mask=None):
         self.geom = geom
         self.fov_mask = fov_mask
         self.exclusion_mask = exclusion_mask
+        self.maps = {}
+
+    def run(self, selection=None):
+        """Make maps.
 
-    def run(self):
-        """tbd
+        Returns dict with keys "counts", "exposure" and "background".
+
+        Parameters
+        ----------
+        selection : list
+            List of str, selecting which maps to make.
+            Available: 'counts', 'exposure', 'background'
+            By default, all maps are made.
         """
+        if selection is None:
+            selection = {'counts', 'exposure', 'background'}
+
+        if 'counts' in selection:
```

Comment:
A pythonic way to avoid the series of `if` could be to do:
```
self.methods = dict(
       counts = self._make_counts,
      exposure = self._make_exposure,
      background = self._make_background
)

for map_type in selection:
       self.maps[map_type] = self.methods(map_type)()

```